### PR TITLE
I fixed ThingIFAPI and its builder to adapt trait.

### PR DESCRIPTION
**Pull request by dairyo**
***This PR is not buildable but this PR is not merged to develop. Please feel free to merge this.***

### Main change

- I fixed ThingIFAPI and ThingIFAPIBuilder to adapt trait.

### Minor change

I commented out implementation of some methods to remove build error. We should fix implementation in another PR.

Related PR: #252

### CODE REVIEW

File: ThingIFSDK/ThingIFSDK/ThingIFAPI.swift [hunk 1]
```diff
@@ -6,17 +6,33 @@
 import Foundation
 
 /** Class provides API of the ThingIF. */
-open class ThingIFAPI: NSObject, NSCoding {
+open class ThingIFAPI<ConcreteAlias: Alias>: NSObject, NSCoding {
+
+    private class var SHARED_NSUSERDEFAULT_KEY_INSTANCE: String {
```
Comment (yongpingchen) [inline ThingIFSDK/ThingIFSDK/ThingIFAPI.swift]: why use `private class var` instead of `private var`?

Comment (dairyo) [reply ThingIFSDK/ThingIFSDK/ThingIFAPI.swift]: @yongpingchen 
The value is constant. so it is no need to instance property.

I changed class property to static property. I think this is more clear.

**Comment by dairyo** [comment]
@yongpingchen I fixed this PR. Could you check this?

**Comment by yongpingchen** [comment]
LGTM

**Pull request merged**